internal/rpcserver/rpcserver.go

```diff
@@ -4954,45 +4977,95 @@ func (s *rpcServer) writeHTTPResponseHeaders(req *http.Request, headers http.Hea
 }
 
 // shutdown terminates the processes of the rpc server.
-func (s *rpcServer) shutdown() error {
-	rpcsLog.Warnf("RPC server shutting down")
+func (s *Server) shutdown() error {
+	log.Warnf("RPC server shutting down")
 	for _, listener := range s.cfg.Listeners {
 		err := listener.Close()
 		if err != nil {
-			rpcsLog.Errorf("Problem shutting down rpc: %v", err)
+			log.Errorf("Problem shutting down rpc: %v", err)
 			return err
 		}
 	}
 	s.wg.Wait()
-	rpcsLog.Infof("RPC server shutdown complete")
+	log.Infof("RPC server shutdown complete")
 	return nil
 }
 
 // RequestedProcessShutdown returns a channel that is sent to when an
 // authorized RPC client requests the process to shutdown.  If the request can
 // not be read immediately, it is dropped.
-func (s *rpcServer) RequestedProcessShutdown() <-chan struct{} {
+func (s *Server) RequestedProcessShutdown() <-chan struct{} {
 	return s.requestProcessShutdown
 }
 
 // NotifyNewTransactions notifies both websocket and getblocktemplate long
 // poll clients of the passed transactions.  This function should be called
 // whenever new transactions are added to the mempool.
-func (s *rpcServer) NotifyNewTransactions(txns []*dcrutil.Tx) {
+func (s *Server) NotifyNewTransactions(txns []*dcrutil.Tx) {
 	for _, tx := range txns {
 		// Notify websocket clients about mempool transactions.
 		s.ntfnMgr.NotifyMempoolTx(tx, true)
 	}
 }
 
+// NotifyWork passes new mining work to the notification manager
+// for block notification processing.
+func (s *Server) NotifyWork(templateNtfn *mining.TemplateNtfn) {
+	s.ntfnMgr.NotifyWork(templateNtfn)
+}
+
+// NotifyStakeDifficulty notifies websocket clients that have registered for
+// maturing ticket updates.
+func (s *Server) NotifyStakeDifficulty(stnd *StakeDifficultyNtfnData) {
+	s.ntfnMgr.NotifyStakeDifficulty(stnd)
+}
+
+// NotifyNewTickets passes a new ticket data for an incoming block from the best
```

Comment:
I almost don't want to nit pick since this PR is looking really good overall now, but, given these are exported and thus will be part of the package API documentation, these really should describe the actual behavior the caller can expect versus what it is internally doing.

Namely, it notifies websocket clients that have registered for maturing ticket updates.

The same goes for all of these notification functions (except they naturally change based on what the client registered for).

I see that, unfortunately, the comments on the related notification manager methods aren't really doing this either, but since they aren't exported, it doesn't matter as much.

Reply:
Good point.  I modified [this commit](https://github.com/decred/dcrd/pull/2288/commits/a902e1d910570eace90f2f9c6f6b5b2b4c8779df) to add more appropriate comments from the perspective of the caller.